### PR TITLE
refactor: unify schema cache flow and drop non-unix daemon fallback

### LINKED ISSUE #141: bug(daemon/mcp): help path should prefer schema cache before live list_tools
## Problem
`uxc <mcp-endpoint> -h` / `<link> -h` (MCP HostHelp) and `uxc <mcp-endpoint> <operation> -h` (OperationHelp) do not consistently use schema cache first.

In daemon path, MCP help handling currently calls live `list_tools()` on stdio/http sessions, which can block when upstream MCP server/tooling is slow or hanging (e.g. Playwright MCP on certain URLs). This violates the expectation that `-h` can be served from cached schema metadata when available.

## Repro
1. Ensure MCP schema cache exists for endpoint (e.g. run one successful help/invoke first).
2. Make upstream MCP endpoint slow/hanging.
3. Run:
   - `playwright-mcp-cli -h`
   - `playwright-mcp-cli browser_navigate -h`
4. Observe help may still block on live MCP calls instead of returning from cache.

## Expected
- For MCP `HostHelp` and `OperationHelp`, daemon should first attempt to serve from schema cache (`tools` in cached schema).
- If cache hit exists (including stale-allowed fallback policy where appropriate), return help immediately without live MCP roundtrip.
- Only fall back to live MCP `list_tools()` when cache miss or cached schema lacks required tool metadata.

## Scope
- Daemon runtime MCP path (`RuntimeAction::HostHelp` / `OperationHelp`).
- Keep execute path behavior unchanged.

## Acceptance Criteria
- With a valid cached schema, `-h` and `<operation> -h` for MCP return quickly even when live MCP endpoint is slow/hanging.
- Behavior remains unchanged for cache miss (still supports live discovery).
- Add regression tests to verify cache-first help for MCP.

## Notes
Related observation: global daemon lock-stall issue was fixed separately; this issue is specifically about MCP help-path cache usage semantics.

### PR DESCRIPTION
**Pull request by jolestar**
## What
Refactor the invoke/help pipeline to make schema cache the single source of truth for MCP/gRPC help, and remove `not(unix)` daemon fallback to enforce the daemon-first architecture.

## Why
Issue #141 exposed that help still depended on backend MCP listing in some paths, which broke the original cache-first design goal.

Closes #141

## How
- Route MCP `HostHelp`/`OperationHelp` through adapter `invoke_with_adapter` path, not daemon MCP session tool listing.
- Keep MCP `Execute` on daemon session reuse path, but add schema-based preflight validation and remove execute-time `tools/list` dependency.
- Refactor MCP adapter help/list/describe to derive from cached schema first, with one controlled refresh path.
- Align gRPC `list_operations`/`describe_operation` to schema-derived flow.
- Remove non-unix runtime fallback in daemon client path; return explicit guidance to run in WSL.
- Update CI/release/docs to match WSL-first Windows policy.
- Add regression scenarios/tests for “tools/list fails after first call”.

## Changes

- [x] MCP help path no longer depends on live `tools/list`
- [x] MCP execute path no longer relists tools on reused daemon sessions
- [x] gRPC help/describe switched to schema-derived flow
- [x] Removed `not(unix)` daemon fallback and clarified platform guidance
- [x] Added regression tests for cache-first help/execute behavior
- [x] Updated CI/release/docs for Windows via WSL

### Unit Tests
- [x] All existing tests pass
- [x] New tests added (if applicable)
- [ ] `cargo test` passes

### Manual Testing
- [ ] Tested against real APIs
- [x] Tested edge cases
- [x] Tested error conditions

### Test Results
```bash
cargo fmt
cargo check
cargo test --test local_e2e_test test_mcp_http_help_uses_cached_schema_when_tools_list_fails_after_first -- --test-threads=1
cargo test --test local_e2e_test test_mcp_http_execute_does_not_depend_on_tools_list -- --test-threads=1
cargo test --test daemon_reuse_test mcp_stdio_execute_does_not_relist_tools_on_reused_session -- --test-threads=1
```

## Checklist

- [x] Code formatted with `cargo fmt`
- [ ] No clippy warnings (`cargo clippy -- -D warnings`)
- [ ] All tests pass (`cargo test`)
- [x] Documentation updated
- [x] Commit messages follow convention
- [x] PR title follows convention (e.g., `feat: add GraphQL support`)

## Breaking Changes

- Native non-unix daemon runtime fallback was removed.
- Windows users should run `uxc` inside WSL for daemon-based workflow.

## Additional Notes

This PR intentionally avoids a localized “help patch” and instead consolidates cache behavior at architecture level for help/list/describe/execute preflight consistency.

### CODE REVIEW

File: src/adapters/grpc.rs [hunk 1]
```diff
@@ -1025,10 +1080,10 @@ impl Adapter for GrpcAdapter {
...
             }],
             return_type: Some(output_type),
-            input_schema: Some(Self::build_operation_input_schema(
-                &descriptors,
-                &input_type,
-            )),
+            input_schema: Some(serde_json::json!({
+                "kind": "grpc_message",
+                "message_type": Self::normalize_type_name(&input_type)
+            })),
```
[the remark this answers is not in the document]
Comment (jolestar) [reply src/adapters/grpc.rs]: Fixed in 22e923d. `describe_operation` now restores the prior `input_schema` shape by reading `method.input_schema` from cached schema (which is now populated during `fetch_schema`), with a backward-compatible fallback when absent.
